build/lpc55pins/src/lib.rs

```diff
@@ -183,21 +256,33 @@ pub fn codegen(pins: Vec<PinConfig>) -> Result<()> {
         }
         match p.name {
             None => (),
-            Some(name) => {
+            Some(ref name) => {
                 let pin = p.pin.get_port_pin();
                 writeln!(&mut buf, "#[allow(unused)]")?;
                 writeln!(
                     &mut buf,
                     "const {}: Pin = Pin::PIO{}_{};",
-                    name, pin.0, pin.1
+                    &name, pin.0, pin.1
                 )?;
+
+                let mut ignore = 0u8;
+                let slot = p.get_pint_slot(&mut ignore);
+                if slot != PintSlot::None {
+                    writeln!(&mut buf, "#[allow(unused)]")?;
+                    writeln!(
+                        &mut buf,
+                        "pub const {}_PINT_MASK: u32 = 1 << {};",
```

Comment:
Passing the raw mask around doesn't seem right and you have a comment below about wanting to validate the mask. Can we pass the `PintSlot` enum around instead? That should always be valid.